util/src/main/java/org/killbill/billing/util/cache/NoOpCacheController.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright 2020-2021 Equinix, Inc
+ * Copyright 2014-2021 The Billing Project, LLC
+ *
+ * The Billing Project licenses this file to you under the Apache License, version 2.0
+ * (the "License"); you may not use this file except in compliance with the
+ * License.  You may obtain a copy of the License at:
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.  See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.killbill.billing.util.cache;
+
+import java.util.List;
+
+import org.killbill.billing.util.cache.Cachable.CacheType;
+
+import com.google.common.base.Function;
+import com.google.common.collect.ImmutableList;
+
+// No support to turn off a cache in Ehcache 3 / JCache (JSR 107) unfortunately
+public class NoOpCacheController<K, V> implements CacheController<K, V> {
+
+    private final BaseCacheLoader<K, V> baseCacheLoader;
+
+    public NoOpCacheController(final BaseCacheLoader<K, V> baseCacheLoader) {
+        this.baseCacheLoader = baseCacheLoader;
+    }
+
+    @Override
+    public List<K> getKeys() {
+        return ImmutableList.<K>of();
+    }
+
+    @Override
+    public boolean isKeyInCache(final K key) {
+        return false;
+    }
+
+    @Override
+    public V get(final K key, final CacheLoaderArgument cacheLoaderArgument) {
```

Comment:
I don't understand fully the impl. of this class, and why for instance it is not enough to return `null` (nothing in the cache for the key)

Reply:
This isn't the cache: this is the `CacheController`. Because we have implemented the Read-through pattern (https://www.ehcache.org/documentation/3.3/caching-patterns.html#read-through), we never really see the caches directly, but instead a `CacheController` (which knows when to delegate to the loader).